animation_nodes/nodes/generic/mix_data.py

```diff
@@ -25,27 +45,73 @@ class MixDataNode(bpy.types.Node, AnimationNode):
         description = "Clamp factor between 0 and 1",
         default = False, update = executionCodeChanged)
 
+    useAList: VectorizedSocket.newProperty()
+    useBList: VectorizedSocket.newProperty()
+    useFactorList: VectorizedSocket.newProperty()
+
     def create(self):
-        self.newInput("Float", "Factor", "factor")
-        self.newInput(self.dataType, "A", "a")
-        self.newInput(self.dataType, "B", "b")
-        self.newOutput(self.dataType, "Result", "result")
+        self.newInput(VectorizedSocket("Float", "useFactorList", ("Factor", "factor"),
+            ("Factors", "factor")))
+        self.newInput(VectorizedSocket(self.dataType, ["useAList"], ("A", "a"), ("As", "a")))
+        self.newInput(VectorizedSocket(self.dataType, ["useBList"], ("B", "b"), ("Bs", "b")))
+
+        self.newOutput(VectorizedSocket(self.dataType, ["useAList", "useBList", "useFactorList"],
+            ("Result", "result"), ("Results", "results")))
 
     def draw(self, layout):
         layout.prop(self, "clampFactor")
 
     def drawLabel(self):
+        if any([self.useAList, self.useBList, self.useFactorList]):
+            return nodeTypes[self.outputs[0].dataType.strip("List").rstrip()]
         return nodeTypes[self.outputs[0].dataType]
 
-    def getExecutionCode(self, required):
-        if self.clampFactor:
-            yield "f = min(max(factor, 0.0), 1.0)"
+    def getExecutionFunctionName(self):
+        if any([self.useAList, self.useBList, self.useFactorList]):
+            return "execute_MixDataList"
         else:
-            yield "f = factor"
-        yield getMixCode(self.dataType, "a", "b", "f", "result")
-
-def getMixCode(dataType, mix1 = "a", mix2 = "b", factor = "f", result = "result"):
-    if dataType in ("Float", "Vector", "Quaternion"): return "{} = {} * (1 - {}) + {} * {}".format(result, mix1, factor, mix2, factor)
-    if dataType == "Matrix": return "{} = {}.lerp({}, {})".format(result, mix1, mix2, factor)
-    if dataType == "Color": return "{} = [v1 * (1 - {}) + v2 * {} for v1, v2 in zip({}, {})]".format(result, factor, factor, mix1, mix2)
-    if dataType == "Euler": return "{} = animation_nodes.utils.math.mixEulers({}, {}, {})".format(result, mix1, mix2, factor)
+            return "execute_MixData"
+
+    def execute_MixData(self, factor, a, b):
+        f = factor
+        if self.clampFactor:
+            f = min(max(factor, 0.0), 1.0)
+        return getMixCode(self.dataType, a, b, f)
+
+    def execute_MixDataList(self, factor, mix1, mix2):
+        if self.dataType == "Float":
+            factors = VirtualDoubleList.create(factor, 0)
+            mix1s = VirtualDoubleList.create(mix1, 0)
+            mix2s = VirtualDoubleList.create(mix2, 0)
```

Comment:
Use the `createMultiple` method to shorten this.

Reply:
Done.